src/lightning/pytorch/loops/optimization/automatic.py

```diff
@@ -59,19 +59,20 @@ def _clone_loss(self) -> None:
     def from_training_step_output(cls, training_step_output: STEP_OUTPUT, normalize: int = 1) -> "ClosureResult":
         closure_loss, extra = None, {}
 
-        if isinstance(training_step_output, dict):
+        if isinstance(training_step_output, Mapping):
             closure_loss = training_step_output.get("loss")
             if closure_loss is None:
                 raise MisconfigurationException(
-                    "In automatic_optimization, when `training_step` returns a dict, the 'loss' key needs to be present"
+                    "In automatic_optimization, when `training_step` returns a Mapping, "
+                    "the 'loss' key needs to be present"
                 )
             extra = {k: v for k, v in training_step_output.items() if k != "loss"}
         elif isinstance(training_step_output, Tensor):
             closure_loss = training_step_output
         elif training_step_output is not None:
             raise MisconfigurationException(
                 "In automatic optimization, `training_step` must return a Tensor, "
-                "a dict, or None (where the step will be skipped)."
+                "a Mapping, or None (where the step will be skipped)."
```

Comment:
```suggestion
                "a Mapping (dict), or None (where the step will be skipped)."
```
(a beginner might not know hat a mapping is)